pyuvsim/telescope.py

```diff
@@ -128,6 +128,31 @@ def __init__(self, beam_list=None, uvb_params=None):
         self.uvb_params.update(list_uvb_params)
         self.uvb_params.update(uvb_params)    # Optional parameters for the UVBeam objects
 
+        x_orientation_list = []
+        for beam in beam_list:
+            if self.string_mode:
+                beam_use = self._str_to_obj(beam)
+            else:
+                beam_use = beam
+
+            if hasattr(beam_use, "x_orientation"):
+                x_orientation_list.append(beam_use.x_orientation)
+        if (
+            len(x_orientation_list) == 0
+            or x_orientation_list.count(None) == len(x_orientation_list)
```

Comment:
is it possible to have `None` for `x_orientation`? It seems to me that a null value should be _either_ be encoded as it not existing _or_ being `None`, but not both?

Reply:
UVBeams (and possibly polarized analytic beams in the future) always have the x_orientation attribute and it is either set to `None` or to `east` or `north`. Unpolarized analytic beams do not have the x_orientation attribute and they don't need it because the two polarizations are identical. If a user is only using unpolarized analytic beams I didn't think they should get a warning.

Reply:
hmmm, not sure I am fully on board with having different API's for the UVBeam/AnalyticBeam. I'd prefer the API to be the same and just to have `None` signify that the parameter doesn't exist. But since that is not the case, I'm happy with this code.

Reply:
We can discuss this in future, we're planning to move AnalyticBeam to pyuvdata to better support it. I worry that there are two different meanings of None in that case -- One is that it's not defined but the two pols are different, the other is that there is no difference between the pols so it doesn't matter. Of course any realistic beam will have differences in the pol responses but I'm not sure we want to warn if the users are using beams that have no intrinsic polarization.

Reply:
I see what you're saying. You might be right -- let's table the discussion for when `AnalyticBeam` moves to `pyuvdata`.